.github/copilot-instructions.md

```diff
@@ -0,0 +1,177 @@
+# Overview
+
+This repo contains a collection of Kubernetes Custom Resource Definitions and their controllers. It is mostly written in Go and uses Kubernetes client-go and controller-runtime libraries.
+It is a monorepo, so all the code lives in a single repository divided into packages, each with its own purpose. 
+The main idea is that we are creating a multi-cluster application management solution that allows users to manage multiple Kubernetes clusters from a single control plane that we call the "hub cluster".
+
+## General Rules
+
+- Use @terminal when answering questions about Git.
+- If you're waiting for my confirmation ("OK"), proceed without further prompting.
+- Follow the [Uber Go Style Guide](https://github.com/uber-go/guide/blob/master/style.md) if possible.
```

Comment:
should we also include these?  
https://go.dev/wiki/Style

checked the uber style, which only includes the [Go Code Review Comments](https://go.dev/wiki/CodeReviewComments)